app/misc/authenticate.ts

```diff
@@ -20,18 +20,69 @@ var repoNotFound = 0;
 var signed = 0;
 var changes = 0;
 let signedAfter = false;
-let loginScopes = [
-  "repo",
-  "user"
-];
 
+let oauthpass = 'x-oauth-basic';
+let electronOauth2 = require('electron-oauth2');
+let account;
+
+//config from oauth library
+const OauthConfig = {
+  clientId: 'c2509d8769f5f1e46028',
+  clientSecret: 'e4717832659e95c2f8f4237a3390cf09013b94f7',
+  authorizationUrl: 'https://github.com/login/oauth/authorize',
+  tokenUrl: 'https://github.com/login/oauth/access_token',
+  useBasicAuthorizationHeader: false,
+  redirectUri: 'http://localhost:8080/user/signin/callback'
+};
+
+const windowParams = {
+  alwaysOnTop: true,
+  autoHideMenuBar: true,
+  webPreferences: {
+    nodeIntegration: false,
+  }
+};
+const options = {
+  scope: 'repo user',
+  accessType: 'admin script'
+};
+const myApiOauth = electronOauth2(OauthConfig, windowParams);
+
+function authenticateUser(callback) {
+  // use oauth call to get token
+  myApiOauth.getAccessToken(options)
+    .then(token => {
+      if(!token) {
+        return;
+      }
+      storeOauthToken(token['access_token']);
+      client = github.client(token['access_token']);
+      if (!client.token) {
+        return;
+      }
+      //grab client information from body object that contains user name.
+      client.get('/user', {}, function (err, status, body, headers) {
+        account = body;
+        //after having the user information, start to sign in
+        getUserInfo(callback);
+      });
 
+    }, err => {
+      console.log('cannot obtain token', err);
+	}).catch( err => {
+    console.log(err);
+  });
+}
 
-//constructor method to create a new copy of cred every time
-function createCredentials(username, password) {
-  this.username = username;
-  this.password = password;
-  this.credentials = Git.Cred.userpassPlaintextNew(username,password);
+//Called then user pushes to sign out even if they have commited changes but not pushed; prompts a confirmation modal
+function CommitNoPush() {
+        if (CommitButNoPush == 1) {
```

Comment:
We have removed CommitButNoPush and replaced it with `getAheadBehindCommits()`.

Please change this to not use the global variable `commitButNoPush`